lxa_iobus/node.py

```diff
@@ -125,7 +125,8 @@ async def sdo_read(self, index, sub_index, timeout=DEFAULT_TIMEOUT):
 
             # We get a packet where the size field is used
             if response.readable_transfer_type == "DataWithSize":
-                return response.data[0 : 4 - response.number_of_bytes_not_used]
+                size = 4 - response.number_of_bytes_not_used
+                return response.data[0:size]
```

Comment:
What problem reported by `flake` does this fix? For me this just seems to be a more complex way of expressing the same thing.

Reply:
It stops `black` and `flake8` fighting over the formatting, because black inserts whitespace when doing arithmetic while indexing a range, but flake complains about whitespaces there:

Format by hand:

```python
return response.data[0:4 - response.number_of_bytes_not_used]
```

Run black:

```python
return response.data[0 : 4 - response.number_of_bytes_not_used]
```

Run flake:

```bash
$ flake8 .
./node.py:10:27: E203 whitespace before ':'
```

So it is a workaround. We could also configure flake to ignore this error kind, but I would rather keep the black/flake config minimal.